crates/kitsune_p2p/kitsune_p2p/src/types/event.rs

```diff
@@ -39,10 +41,38 @@ pub struct SignNetworkDataEvt {
     pub data: Arc<Vec<u8>>,
 }
 
+#[derive(Debug)]
+/// Store the AgentInfo as signed by the agent themselves.
+pub struct PutAgentInfoSignedEvt {
+    /// The "space" context.
+    pub space: Arc<super::KitsuneSpace>,
+    /// The "agent" context.
+    pub agent: Arc<super::KitsuneAgent>,
+    /// The signed agent info.
+    pub agent_info_signed: AgentInfoSigned,
+}
+
+#[derive(Debug)]
+/// Get agent info as previously signed and put.
+pub struct GetAgentInfoSignedEvt {
+    /// The "space" context.
+    pub space: Arc<super::KitsuneSpace>,
+    /// The "agent" context.
+    pub agent: Arc<super::KitsuneAgent>,
+    /// The signed agent info key.
+    pub agent_info_signed_key: AgentInfoSignedKey,
```

Comment:
I'm kind of okay with space and agent being duplicated in both inside and outside the AgentInfo in PutAgentInfoSignedEvt, because we need them inside for the signature to make sense, and for symmetry with all the other kitsune events that gave space and agent on the outside.

However, I don't think we need/want this agent_info_signed_key here. It is duplicating the data of space and agent right next to it, and it is particular to our LMDB implementation... other implementations may choose to key their data differently. I think AgentInfoSignedKey should be defined in the conductor agent_store file, and not used / passed in here.